File: Golang/ex9.go
package main

import (
	"fmt"
)

// Interface Imprimivel
type Imprimivel interface {
	Imprimir()
}

type Relatorio struct{}

func (r Relatorio) Imprimir() {
	fmt.Println("Imprimindo o Relatório.")
}

type Contrato struct{}

func (c Contrato) Imprimir() {
	fmt.Println("Imprimindo o Contrato.")
}

func main() {
	var relatorio Imprimivel = Relatorio{}
	var contrato Imprimivel = Contrato{}

	relatorio.Imprimir()
	contrato.Imprimir()
}

File: Java/ex9.java
interface Imprimivel {
    void imprimir();
}

class Relatorio implements Imprimivel {
    @Override
    public void imprimir() {
        System.out.println("Imprimindo o Relatório.");
    }
}

class Contrato implements Imprimivel {
    @Override
    public void imprimir() {
        System.out.println("Imprimindo o Contrato.");
    }
}

public class Main {
    public static void main(String[] args) {
        Imprimivel relatorio = new Relatorio();
        Imprimivel contrato = new Contrato();

        relatorio.imprimir();
        contrato.imprimir();
    }
}
